### PR TITLE
Adding a whole bunch of css

### PR DESCRIPTION
**Pull request by mbrandonw**
just stuff we need for grids and style guides and what not.

one of the cool things is the implementation of `calc`, check out this test:

```swift
  func testCalc() {
    let css: Css = img % (
      width(pct(25) - px(20))
        <> margin(all: px(10))
    )

    XCTAssertEqual(
      """

img {
  width  : -webkit-calc(25.0% - 20.0px);
  width  : -moz-calc(25.0% - 20.0px);
  width  : -ms-calc(25.0% - 20.0px);
  width  : -o-calc(25.0% - 20.0px);
  width  : calc(25.0% - 20.0px);
  margin : 10.0px 10.0px 10.0px 10.0px;
}


""",
      render(config: pretty, css: css)
    )
  }
```

### CODE REVIEW

File: Sources/CSS/Geometry.swift [hunk 1]
```diff
@@ -75,9 +75,16 @@ public func margin(all: Size) -> Css {
...
-    bottom
+    .init(.plain("margin")),
+    Margin(top: topBottom, right: leftRight, bottom: topBottom, left: leftRight)
   )
 }
+
+public let width: (Size) -> Css = key("width")
+public func maxWidth(_ s: Size) -> Css {
+  return s |> key("max-width")
+}
```
Comment (mbrandonw) [inline Sources/CSS/Geometry.swift]: so we have two `mixWidth`s that are overloaded on their return type. at first one was defined as a `let` closure and the other a function. apparently there is some kind way that one is preferred over the other cause Swift couldn't do type inference on those two things. they have to both be functions

File: Tests/CSSTests/ResetTests.swift [hunk 2]
```diff
@@ -7,8 +7,7 @@ class ResetTests: XCTestCase {
     XCTAssertEqual(
       """
 
-a, abbr, acronym, address, article, aside, audio, b, blockquote, body, canvas, caption, cite, code, dd, details, div, dl, dt, em, embed, fieldset, figure, figcaption, footer, form, h1, h2, h3, h4, h5, h6, header, hgroup, html, iframe, i, img, input, label, legend, li, menu, nav, ol, section, span, strong, summary, table, tbody, td, tfoot, th, thead, time, tr, p, pre, q, u, ul, video
-{
+a, abbr, acronym, address, article, aside, audio, b, blockquote, body, canvas, caption, cite, code, dd, details, div, dl, dt, em, embed, fieldset, figure, figcaption, footer, form, h1, h2, h3, h4, h5, h6, header, hgroup, html, iframe, i, img, input, label, legend, li, menu, nav, ol, section, span, strong, summary, table, tbody, td, tfoot, th, thead, time, tr, p, pre, q, u, ul, video {
```
Comment (mbrandonw) [inline Tests/CSSTests/ResetTests.swift]: I just decided it's better to have `{` on the same line as the selector...

File: Tests/CSSTests/FullStylesheetTests.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import XCTest
+import Prelude
+import CSS
+
+class FullStylesheetTests: XCTestCase {
```
Comment (mbrandonw) [inline Tests/CSSTests/FullStylesheetTests.swift]: I wanna keep adding on to this to make one big ole stylesheet to test with. starting with something simple for now

File: Sources/CSS/Box.swift [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  public static var inherit: BoxType {
+    return .init(boxType: inheritValue)
+  }
+}
+
+extension BoxType: ExpressibleByStringLiteral {
+  public init(stringLiteral value: String) {
+    self = .init(boxType: .init(stringLiteral: value))
```
Comment (stephencelis) [inline Sources/CSS/Box.swift]: Do we need the inner `.init(stringLiteral: value)` here? Can it just be `.init(boxType: value)`?

File: Sources/CSS/CssSelector.swift [hunk 5]
```diff
@@ -234,3 +237,7 @@ public func & (lhs: CssSelector, rhs: CssSelector) -> CssSelector {
   return .combined(lhs, rhs)
 }
 
+infix operator ~: infixr6
```
Comment (stephencelis) [inline Sources/CSS/CssSelector.swift]: 😄

File: Sources/CSS/Text.swift [hunk 6]
```diff
@@ -33,3 +33,44 @@ public struct Quotes: Val, None {
...
+  }
+
+  public static var center: TextAlign {
+    return .init(centerValue)
+  }
+}
+
+extension TextAlign: ExpressibleByStringLiteral {
+  public init(stringLiteral value: String) {
+    self = .init(.init(stringLiteral: value))
```
Comment (stephencelis) [inline Sources/CSS/Text.swift]: Same here: inner-nest necessary?

File: Sources/CSS/Text.swift [hunk 7]
```diff
@@ -33,3 +33,44 @@ public struct Quotes: Val, None {
...
+  }
+}
+
+public let justify: TextAlign = "justify"
+public let matchParent: TextAlign = "match-parent"
+public let start: TextAlign = "start"
+public let end: TextAlign = "end"
+
+public let textAlign: (TextAlign) -> Css = key("text-align")
```
Comment (stephencelis) [inline Sources/CSS/Text.swift]: Wondering when, in the interest of avoiding function overload collisions, we should avoid `let`s.

Comment (mbrandonw) [inline Sources/CSS/Text.swift]: yup, the `let`s are nice and short, but not worth it.

File: Sources/CSS/Box.swift (hunk 4 above)
Comment (mbrandonw) [inline Sources/CSS/Box.swift]: it can't cause `value` is a `String` and not a literal.

the literal stuff is kinda weird.

**Pull request merged**